### PR TITLE
Update the build.release task

### LINKED ISSUE #1387: The lucky build task should emit to bin
We added in https://github.com/luckyframework/lucky/pull/1358 which builds `start_server` in to `bin`, but missed this task which still emits the binary to the root directory.

https://github.com/luckyframework/lucky/blob/5f8a9d1836a3fbeb40b4570ec2f44c0f47d4522c/tasks/build/release.cr#L13

### PR DESCRIPTION
**Pull request by jwoertink**
## Purpose
Fixes #1387

## Description
We should now be building all binaries to the `bin/` dir. When we updated to do this, the `build.release` task was missed. This also uses the `output` property from LuckyCli.

One strange thing I couldn't figure out is that `log "Building binary with '#{command}"` prints out to the screen now 🤔  Even though it's a `IO::Memory`, it still seems to print just that one message. It doesn't print out the other one... 

## Checklist
* [x] - An issue already exists detailing the issue/or feature request that this PR fixes
* [x] - All specs are formatted with `crystal tool format spec src`
* [x] - Inline documentation has been added and/or updated
* [x] - Lucky builds on docker with `./script/setup`
* [x] - All builds and specs pass on docker with `./script/test`

### CODE REVIEW

File: tasks/build/release.cr [hunk 1]
```diff
@@ -6,20 +6,19 @@ class Build::Release < LuckyCli::Task
 
   summary "Compile app for production"
 
-  def initialize(@io : IO = STDOUT, @error_io : IO = STDERR)
-  end
+  property error_io : IO = STDERR
```
Comment (jwoertink) [inline tasks/build/release.cr:9]: This should be added in to LuckyCli::Task directly to match up with `output`.

File: tasks/build/release.cr [hunk 2]
```diff
@@ -6,20 +6,19 @@ class Build::Release < LuckyCli::Task
...
 
   def call
-    command = "crystal build --release src/start_server.cr"
+    command = "crystal build --release src/start_server.cr -o bin/start_server"
 
     log "Building binary with '#{command}"
```
Comment (jwoertink) [inline tasks/build/release.cr:14]: Why does this print out, but the one below doesn't?

Screen Shot 2021-04-02 at 2 46 41 PM

Comment (matthewmcgarvey) [reply tasks/build/release.cr:14]: Maybe try adding an else and check if the process isn't completing successfully?

Comment (jwoertink) [reply tasks/build/release.cr:14]: ah. Good call. It is failing.. 🤔  but I guess why.

Comment (jwoertink) [reply tasks/build/release.cr:14]: oh, but that still doesn't explain why `IO::Memory` is printing to STDOUT 🤨

Comment (matthewmcgarvey) [reply tasks/build/release.cr:14]: Ohh, it's actually the second spec that is printing to stdout, it doesn't have it's output changed, only error_io

Comment (jwoertink) [reply tasks/build/release.cr:14]: ooohhhh 😂  yup. That fixes it. Thanks!

**Review by matthewmcgarvey — APPROVED** [review]

**Pull request merged**